server/src/main/java/org/opensearch/ingest/IngestService.java

```diff
@@ -107,6 +108,7 @@ public class IngestService implements ClusterStateApplier, ReportingService<Inge
     public static final String NOOP_PIPELINE_NAME = "_none";
 
     public static final String INGEST_ORIGIN = "ingest";
+    private static final int MAX_PIPELINE_ID_LENGTH = 20;
```

Comment:
@andrross @msfroh I kept the max length as 20. Open to suggestion here

Reply:
But why? I can see having some limit here, so things don't fail in a bad way with like a 5MB ID. But 20 seems incredibly restrictive and likely to cause pain on upgrades. We restrict document IDs to [512 UTF-8 bytes](https://github.com/opensearch-project/OpenSearch/blob/0d54c1600b16081c15902a095111a9a530767099/server/src/main/java/org/opensearch/action/DocWriteRequest.java#L266). I would not recommend restricting this to less than 512, and would be fine with something larger.

Reply:
Also, `String::length` returns the number of unicode code points. Assuming the real limits here are related to the size of the physical representation, restricting to some number of UTF-8 bytes is probably the better way to go.

Reply:
Agreed. Keeping the length consistent across documents length and pipeline id makes sense. Pushed the change

Reply:
Yeah -- I agree that the limit of 512 bytes makes more sense. 

It still lets folks append `-v2-revised-updated-new[2]-backup` suffixes to their pipelines for [versioning](https://github.com/opensearch-project/OpenSearch/issues/17539), but it prevents pipeline names from eating up all of cluster state. (Now we just need to worry about the pipeline definition's contribution to cluster state size.)

Reply:
@msfroh do you mean limiting the number of processors while defining the pipeline to avoid overloading the cluster state?

Reply:
I think @msfroh's point is that cluster state isn't the ideal store for something like pipeline definitions because in theory they can be quite large, which just adds overhead to all operations that involve reading or writing cluster state. The solution to that is some sort of metadata store that isn't cluster state. But that's a bigger issue separate from this.